src/python/WMCore/ReqMgr/Service/Auxiliary.py

```diff
@@ -317,11 +318,34 @@ class UnifiedConfig(AuxBaseAPI):
     def setName(self):
         self.name = "UNIFIED_CONFIG"
 
+
 class TransferInfo(AuxBaseAPI):
     """
     Handle information related to the automatic transfers made
     by the MicroServices.
     """
 
     def setName(self):
-        self.name = "TRANSFER"
\ No newline at end of file
+        self.name = "TRANSFER"
+
+
+class ParentLocks(AuxBaseAPI):
+    """
+    Handle list of locked parent datasets
+    """
+
+    def setName(self):
+        self.name = "PARENT_LOCKS"
+
+    @restcall(formats=[('text/plain', PrettyJSONFormat()), ('application/json', JSONFormat())])
+    def get(self, subName=None):
+        """
+        Return entire self.name document
+        subName is subcategory of document which is added as postfix string
+        """
+        try:
+            sw = self.reqmgr_aux_db.document(self.name)
+        except CouchNotFoundError:
+            raise NoSuchInstance
+
+        return rows(sw['parentlocks'])
```

Comment:
I might need a check to see if this 'parentlocks' key exists to be safe.

Reply:
If we want to simply return a list of datasets and not the whole couch document, it is not obvious to me how to do that with the get method from the super class.

Reply:
I think it's fine to return the whole document. Even better if the document contains a `timestamp` field, such that one can know when the document is stale.